### PR TITLE
docs: add pre-release notice and review deps sections to RELEASE.md

### PR DESCRIPTION
**Pull request by cbeach47**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] documentation is changed or added

##### Description of change
Added pre-release notice and review dependencies sections to RELEASE.md.

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
CI is seeing a weird unrelated error.
```
 org.apereo.portal.concurrency.locking.JpaClusterLockDaoTest > testConcurrentLocking FAILED
    java.lang.AssertionError at JpaClusterLockDaoTest.java:140

373 tests completed, 1 failed, 108 skipped

> Task :uPortal-webapp:test FAILED
```
restarting CI to see if this resolves itself

File: docs/developer/other/RELEASE.md [hunk 1]
```diff
@@ -35,6 +35,30 @@ We encourage performing releases directly from a clone of the official repositor
 
 This means when testing on `uPortal-start` for the release, you should use the `apereo` repository but configure the `uPortalVersion` to be the SNAPSHOT version that you'll build in the following steps.
 
+## Send Pre-Release Notice to Community
+For any non-snapshot release, email a notice to `uportal-dev` a couple of working days prior to cutting the release.
```
Comment (ChristianMurphy) [inline docs/developer/other/RELEASE.md:39]:
```suggestion

## Send Pre-Release Notice to Community
For any non-snapshot release, email a notice to `uportal-dev` a couple of working days prior to cutting the release.
```

File: docs/developer/other/RELEASE.md [hunk 2]
```diff
@@ -35,6 +35,30 @@ We encourage performing releases directly from a clone of the official repositor
...
+Assuming adopters use the community build of uPortal, there are three primary sets of dependencies to be considered:
+- uPortal
+- uPortal-start (community version)
+- uPortal-start (customized)
+
+Ideally, dependencies in all of the above areas should be at the latest versions.  In reality, dependencies may lag, and be out of sync with each other.
+
+### Specific Dependencies
+
+- `nodejsVersion` - uPortal and uPortal-start use NodeJS for building artifacts and Gradle tasks. They are not required to be in sync.
```
Comment (ChristianMurphy) [inline docs/developer/other/RELEASE.md:56]: Ideally they should be on the same major version.
But minor and patch versions don't need to match. 👍

**Review by ChristianMurphy — APPROVED** [review]

File: docs/developer/other/RELEASE.md (hunk 1 above)
Comment (cbeach47) [reply docs/developer/other/RELEASE.md:39]: fixed.

**Review by ChristianMurphy — APPROVED** [review]

**Comment by cbeach47** [comment]
Thanks for the reviews @ChristianMurphy 🎉

**Pull request merged**